Source/WebCore/html/HTMLCanvasElement.cpp

```diff
@@ -1013,6 +1016,25 @@ void HTMLCanvasElement::prepareForDisplay()
     notifyObserversCanvasDisplayBufferPrepared();
 }
 
+void HTMLCanvasElement::dynamicRangeLimitDidChange(PlatformDynamicRangeLimit dynamicRangeLimit)
+{
+    if (m_dynamicRangeLimit == dynamicRangeLimit)
+        return;
+
+    m_dynamicRangeLimit = dynamicRangeLimit;
+#if ENABLE(PIXEL_FORMAT_RGBA16F)
+    if (m_context)
+        m_context->setDynamicRangeLimit(dynamicRangeLimit);
+#endif // ENABLE(PIXEL_FORMAT_RGBA16F)
+}
+
+double HTMLCanvasElement::effectiveDynamicRangeLimitValue() const
```

Comment:
A 'ForTesting' suffix for these functions would be nice, since otherwise the fallback `-2.0` is somewhat confusing.

Reply:
To avoid any trouble: I'll rename the function getContextEffectiveDynamicRangeValue, change the return to `optional<double>`, and the internals will throw an exception if there's no value.